builtins.go

```diff
@@ -318,13 +318,25 @@ func builtinObjectHasEx(e *evaluator, objp potentialValue, fnamep potentialValue
 		return nil, err
 	}
 	for _, fieldname := range objectFields(obj, hidden.value) {
-		if fieldname == fname.value {
+		if fieldname == string(fname.value) {
 			return makeValueBoolean(true), nil
 		}
 	}
 	return makeValueBoolean(false), nil
 }
 
+func builtinPow(e *evaluator, basep potentialValue, expp potentialValue) (value, error) {
+	base, err := e.evaluateNumber(basep)
+	if err != nil {
+		return nil, err
+	}
+	exp, err := e.evaluateNumber(basep)
+	if err != nil {
+		return nil, err
+	}
+	return makeDoubleCheck(e, math.Pow(base.value, exp.value))
```

Comment:
Have you looked into go's math.pow vs the C++ equivalent's behavior? I'd imagine this is where the two jsonnet's implementations might start to exhibit weird differences.

Reply:
Interesting. I couldn't find any obvious differences. Both C++ and Go documentation have long lists of special cases. The lists are different, but most likely equivalent.

I was able to find the implementation of pow in Go (https://github.com/golang/go/blob/master/src/math/pow.go#L13). It references FreeBSD C library and IEEE 754-2008. So I hope there is some standard for what it should do.

I couldn't find any mainstream C++ implementation to compare what it actually does.

Do you have any suggestions for how I can check it more thoroughly?

Reply:
Ha sorry no suggestions, just wondering if this has come up yet. Perhaps we should make a list of weirdness we might expect in the future/should test given the opportunity.